controller/aks-cluster-config-handler.go

```diff
@@ -31,6 +31,7 @@ const (
 	aksConfigActivePhase     = "active"
 	aksConfigUpdatingPhase   = "updating"
 	aksConfigImportingPhase  = "importing"
+	maxLengthPoolName        = 6
```

Comment:
nit: 
```suggestion
	poolNameMaxLength        = 6
```

Reply:
I think we should enforce this in the UI also, if you can make that happen

Reply:
changed the constant name

Reply:
Added to UI issue https://github.com/rancher/rancher/issues/31315#issuecomment-839970611